### ISSUE TITLE
Apple App Clips and Android Instant Apps

### ISSUE THREAD

**Issue by antonioberetini** [capacitor-community/proposals#113]
## Plugin Request

**Name**: App Clips & Instant Apps
**Package**: `@capacitor-community/app-clips-&-instant-apps

### Platform(s)
IOS AND ANDROID

### Existing Solutions
I was not able to find any existing solutions for this use case

### Description
Instant apps and app clips are build to allow users to quickly access applications without having to download them from the stores. These applications are invoked with use of a QR code or via NFC tags. These features are often used in restaurants, bars gas stations or anywhere else where instant app function is required and bandwidth is often limited.

Apple App Clips: https://developer.apple.com/documentation/app_clips

Android Instant Apps: https://support.google.com/googleplay/android-developer/answer/9900182?hl=en

I am happy to provide payment for the integration of these features via Capacitor.

**Comment by johnborges** [comment]
I don't think that App Clips or Instant Apps are features/functionality that a plugin would expose. From reading through the docs, it seems to be more of an App specific configuration.

**Comment by antonioberetini** [comment]
@johnborges yes but capacitor needs to be aware of which portions of the app are meant to be used for instant apps and app clips. Configuration needs to live somewhere for both Ios and Android

**Comment by didier** [comment]
Would really love to see something like this implemented. This to me seems like the perfect candidate for what Capacitor stands for.

**Comment by wokkaflokka** [comment]
In the past (2020), the size limit for a production Instant App was 5MB. This was hard for enterprise apps to achieve in practice and would have made it functionally difficult for a capacitor application to be within the size range.

I see now that the size constraint is now [15 MB.](https://developer.android.com/topic/google-play-instant/guides/reduce-module-size) This is much more achievable and I could see the ability to distribute a capacitor app in this way.

> I don't think that App Clips or Instant Apps are features/functionality that a plugin would expose. From reading through the docs, it seems to be more of an App specific configuration.

This is a correct understanding. To the commentary in this thread, perhaps the project generator could support generating some instant-app specific configuration. Delivering an Instant App has non-trivial product and process implications so it doesn't seem terrible that it requires some thought to get started. But I can also see an argument for enabling this distribution by default -- perhaps that could lead to a better future.

**Comment by niallrr** [comment]
Let's build it.

**Comment by ycocatrix** [comment]
It will not help but I would be very interested in a plugin allowing App Clips and Instant Apps.
Does someone knows if it's planned ?

**Comment by drakedeatonuk** [comment]
Anyone make any progress on this?

**Comment by WcaleNieWolny** [comment]
@antonioberetini Doing Apple App Clips is doable for iOS. Let me explain (I got a working PoC)

1. Create a new App Clip in XCode for an existing Capacitor App
	- You need to click on `File` -> `New` -> `Target`
	- Let's then select `App Clip`
	- Name it. I'll name it 'test clip'. It will become important later
	- Select `interface` to storyboard
	- Disable testing system
	- Final thing should look like this
  
![Image](https://github.com/user-attachments/assets/85c829d3-a35a-45bf-9ffa-43124d88b125)

2. Let's setup signing - you'll need a paid Apple Developer account for this
	- Open `test_clip.entitlements` in any text editor OTHER than XCode
    - Change the file with
		```
		    <array>
		        <string>$(AppIdentifierPrefix)me.wcaleniewolny.test.ionic.vue</string>
		    </array>
		```
 	 - You want to replace `me.wcaleniewolny.test.ionic.vue` with the bundle ID of your main Capacitor app
	 - Try to run the app. It SHOULD build

3. Open `Podfile` in `ios/App/Podfile`. Add
```
target 'Test app clip' do
  capacitor_pods
  # Add specific pods for the app clip
end
```
4. Open XCode again, remove `LaunchScreen` and `Main` from the new target (`test clip`)
5. Copy `LaunchScreen`, `Main`, `public` from `App` (main capacitor) to `test clip`
6. Launch your test clip and enjoy Capacitor !
